website/src/content/docs/docs/howtos/Generate client libraries/07types.mdx

```diff
@@ -166,6 +166,259 @@ Foo foo = new Foo("hello");
+ */
+export interface Point {
+  /**
+   * The horizontal position (increases moving right)
+   */
+  x: number;
+
+  /**
+   * The vertical position (increases moving down)
+   */
+  y: number;
+}
+```
+
+</ClientTabItem>
+<ClientTabItem lang="java" >
+
+```java
+/**
+ * Base class for all shape objects that can be drawn on a canvas.
+ */
+@Fluent
+public final class Shape {
+    /**
+     * The x,y coordinates where the shape will be positioned on the canvas.
+     */
+    private Point position;
+
+    /**
+     * Get the position property: The x,y coordinates where the shape will be positioned on the canvas.
+     *
+     * @return the position value.
+     */
+    public Point getPosition() {
+        return this.position;
+    }
+
+    /**
+     * Set the position property: The x,y coordinates where the shape will be positioned on the canvas.
+     *
+     * @param position the position value to set.
+     * @return the Shape object itself.
+     */
+    public Shape setPosition(Point position) {
+        this.position = position;
+        return this;
+    }
+}
+
+/**
+ * Represents a position in a 2D coordinate system.
+ */
+@Fluent
+public final class Point {
+    /**
+     * The horizontal position (increases moving right).
+     */
+    private float x;
+
+    /**
+     * The vertical position (increases moving down).
+     */
+    private float y;
+
+    /**
+     * Get the x property: The horizontal position (increases moving right).
+     *
+     * @return the x value.
+     */
+    public float getX() {
+        return this.x;
+    }
+
+    /**
+     * Set the x property: The horizontal position (increases moving right).
+     *
+     * @param x the x value to set.
+     * @return the Point object itself.
+     */
+    public Point setX(float x) {
+        this.x = x;
+        return this;
+    }
+
+    /**
+     * Get the y property: The vertical position (increases moving down).
+     *
+     * @return the y value.
+     */
+    public float getY() {
+        return this.y;
+    }
+
+    /**
+     * Set the y property: The vertical position (increases moving down).
+     *
+     * @param y the y value to set.
+     * @return the Point object itself.
+     */
+    public Point setY(float y) {
+        this.y = y;
+        return this;
+    }
+}
+```
+
+</ClientTabItem>
+<ClientTabItem lang="go">
+
+```go
+// Shape - Base class for all shape objects that can be drawn on a canvas
+type Shape struct {
+	// Position - The x,y coordinates where the shape will be positioned on the canvas
+	Position *Point
+}
+
+// Point - Represents a position in a 2D coordinate system
+type Point struct {
+	// X - The horizontal position (increases moving right)
+	X *float32
+
+	// Y - The vertical position (increases moving down)
+	Y *float32
+}
+```
+
+</ClientTabItem>
+</ClientTabs>
+
+The `@clientDoc` decorator can be used on:
+
+- Models
+- Model properties
+- Operations
+- Operation parameters
+- Enums
+- Enum members
```

Comment:
We should also make it available for anything that can have a `@doc` decorator. This includes `Namespace`, `Interface`, `Union` etc. I think the decorator's target is `unknown` which implies it works for all of the above. So, we should need to update this documentation.